packages/parallel/init.lua

```diff
@@ -99,75 +438,196 @@ function package:_init (options)
       parallelPagebreak()
       oldfinish(self_)
    end
 end
 
-function package:registerCommands ()
-   self:registerCommand("sync", function (_, _)
+function package:registerCommands()
+   -- shortcut for \parskip
+   self:registerCommand("parskip", function(options, _)
+      local height = options.height or "12pt plus 3pt minus 1pt"
+      SILE.typesetter:leaveHmode()
+      SILE.typesetter:pushExplicitVglue(SILE.types.length(height))
+   end)
+
+   -- Synchronize frame heights and handle page breaks.
+   self:registerCommand("sync", function(_, _)
       local anybreak = false
       local maxheight = SILE.types.length()
-      SU.debug("parallel", "Trying a sync")
-      allTypesetters(function (_, typesetter)
-         SU.debug("parallel", "Leaving hmode on", typesetter.id)
+
+      -- Check for potential page breaks.
+      allTypesetters(function(_, typesetter)
          typesetter:leaveHmode(true)
-         -- Now we have each typesetter's content boxed up onto the output stream
-         -- but page breaking has not been run. See if page breaking would cause a
-         -- break
          local lines = pl.tablex.copy(typesetter.state.outputQueue)
          if SILE.pagebuilder:findBestBreak({ vboxlist = lines, target = typesetter:getTargetLength() }) then
             anybreak = true
          end
       end)
 
+      -- Perform a page break if necessary.
       if anybreak then
          parallelPagebreak()
          return
       end
 
-      allTypesetters(function (frame, typesetter)
-         calculations[frame].heightOfNewMaterial = SILE.types.length()
-         for i = calculations[frame].mark + 1, #typesetter.state.outputQueue do
-            local thisHeight = typesetter.state.outputQueue[i].height + typesetter.state.outputQueue[i].depth
-            calculations[frame].heightOfNewMaterial = calculations[frame].heightOfNewMaterial + thisHeight
-         end
-         if maxheight < calculations[frame].heightOfNewMaterial then
+      -- Calculate the height of new material for balancing.
+      allTypesetters(function(frame, typesetter)
+         calculations[frame].heightOfNewMaterial = calculateFrameHeight(frame, typesetter)
+         if calculations[frame].heightOfNewMaterial > maxheight then
             maxheight = calculations[frame].heightOfNewMaterial
+            log("Value of maxheight after balancing for frame ", frame, ": ", maxheight)
          end
-         SU.debug(
-            "parallel",
-            frame,
-            ": pre-sync content=",
-            calculations[frame].mark,
-            ", now",
-            #typesetter.state.outputQueue,
-            ", height of material:",
-            calculations[frame].heightOfNewMaterial
-         )
       end)
+
+      -- Add balancing glue to align frame heights.
       addBalancingGlue(maxheight)
+
+      -- Handle parskip (spacing between successive frames).
+      local parskip = SILE.settings:get("document.parskip")
+
+      if not parskip.length then
```

Comment:
When does it occur? document.parskip is always defined no?

Reply:
>  I may be wrong there, but somehow it works for me.

Hm... `not parsikip` is still false when parskip is set to 0pt. It's always false ;)

Reply:
> parskip.length is always nil regardless of whatever the document.parskip is set to.

'Cause the parskip is not a **length** object, it's a **vglue** (= with height/depth/width fields, which are lengths)